data/src/main/java/tech/pegasys/artemis/data/TimeSeriesRecord.java

```diff
@@ -114,62 +120,100 @@ public TimeSeriesRecord(
     this.validators = validators;
   }
 
+  public ArrayList<JsonObject> validatorOutputFields() {
+    ArrayList<JsonObject> validatorOutputFields = new ArrayList<>();
+    Gson gson = new GsonBuilder().create();
+
+    for (ValidatorJoin validator : validators) {
+      try {
+        validatorOutputFields.add(
+            gson.fromJson(
+                mapper.writerFor(ValidatorJoin.class).writeValueAsString(validator),
+                JsonObject.class));
+      } catch (JsonProcessingException e) {
+        System.out.println("Exception - Validator output fields ");
+      }
+    }
+    return validatorOutputFields;
+  }
+
   @Override
-  public String toJSON() {
-    try {
-      return mapper.writerFor(TimeSeriesRecord.class).writeValueAsString(this);
-    } catch (JsonProcessingException e) {
-      throw new IllegalArgumentException(e);
+  public void filterOutputFields(List<String> outputFields) {
+    for (String field : outputFields) {
+      switch (field) {
+        case "index":
+          this.outputFieldMap.put("index", getIndex());
+          break;
+
+        case "slot":
+          this.outputFieldMap.put("slot", getSlot());
+          break;
+
+        case "epoch":
+          this.outputFieldMap.put("epoch", getEpoch());
+          break;
+
+        case "block_root":
+          this.outputFieldMap.put("block_root", getBlock_root());
+          break;
+
+        case "block_body":
+          this.outputFieldMap.put("block_body", getBlock_body());
+          break;
+
+        case "lastFinalizedBlockRoot":
+          this.outputFieldMap.put("lastFinalizedBlockRoot", getLastFinalizedBlockRoot());
+          break;
+
+        case "lastFinalizedStateRoot":
+          this.outputFieldMap.put("lastFinalizedStateRoot", getLastFinalizedStateRoot());
+          break;
+
+        case "block_parent_root":
+          this.outputFieldMap.put("block_parent_root", getBlock_parent_root());
+          break;
+
+        case "validators":
+          this.outputFieldMap.put("validators", validatorOutputFields());
+          break;
+
+        case "validators_size":
+          this.outputFieldMap.put("validators_size", getValidators().size());
+          break;
+
+        case "lastJustifiedBlockRoot":
+          this.outputFieldMap.put("lastJustifiedBlockRoot", getLastJustifiedBlockRoot());
+          break;
+
+        case "lastJustifiedStateRoot":
+          this.outputFieldMap.put("lastJustifiedStateRoot", getLastJustifiedStateRoot());
+          break;
+      }
     }
   }
 
+  @Override
+  public String toJSON() {
+    Gson gson = new GsonBuilder().create();
+    GsonBuilder gsonBuilder = new GsonBuilder();
+    String jsonString = gson.toJson(this.outputFieldMap);
+    JsonObject timeSeriesRecord = gson.fromJson(jsonString, JsonObject.class);
+    Gson customGson = gsonBuilder.setPrettyPrinting().create();
+    return customGson.toJson(timeSeriesRecord);
```

Comment:
i think pretty printing is supposed to be optional.  @dangerousfood isn't that what the `format` option is for?

Reply:
Nah, pretty printing just brings in tabs and new line characters. You're thinking of the formatted output that puts brackets around the whole thing